tests/unit_tests/scheduler/test_openpbs_driver.py

```diff
@@ -326,7 +326,13 @@ async def test_that_qdel_will_retry_and_succeed(
     driver._iens2jobid[0] = 111
     await driver.kill(0)
     assert "TRIED" in Path(bin_path / "script_try").read_text(encoding="utf-8")
-    assert "qdel executed" in Path(bin_path / "qdel_output").read_text(encoding="utf-8")
+    if exit_code == QDEL_JOB_HAS_FINISHED:
+        # the job has been already qdel-ed so no need to retry
+        assert not os.path.exists(bin_path / "qdel_output")
+    else:
+        assert "qdel executed" in Path(bin_path / "qdel_output").read_text(
+            encoding="utf-8"
```

Comment:
Nitpick: Is encoding required? Is the default not `"utf-8"` already?

Reply:
We should use the encoding yes, utf-8 is default on our computers yes, but lets be kind to the world.

See https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/unspecified-encoding.html

Reply:
This was enforced earlier when we had  pylint in CI, now I see that our codebase has deterioted on this after the switch to ruff 😢

Reply:
From the PEP it sounds like it only is a problem on Windows. It's what we in the activism business called praxis in our fight against the Microsoft monopoly!

But, joking aside, it is reasonable to specify `encoding="utf-8"` I guess. I always thought it was defined to be as such in Python 3, now that `str` is unicode.